app/coffee/WebsocketLoadBalancer.coffee

```diff
@@ -69,12 +82,27 @@ module.exports = WebsocketLoadBalancer =
 				clientList = io.sockets.clients(message.room_id)
 				# avoid unnecessary work if no clients are connected
 				return if clientList.length is 0
-				logger.log {channel:channel, message: message.message, room_id: message.room_id, message_id: message._id, socketIoClients: (client.id for client in clientList)}, "distributing event to clients"
+				logger.log {
+					channel: channel,
+					message: message.message,
+					room_id: message.room_id,
+					message_id: message._id,
+					socketIoClients: (client.id for client in clientList)
+				}, "distributing event to clients"
 				seen = {}
-				for client in clientList when not seen[client.id]
-					seen[client.id] = true
-					client.emit(message.message, message.payload...)
+				# Send the messages to clients async, don't wait for them all to finish
+				Async.eachSeries clientList
```

Comment:
Re the description, I can't see why this would be significantly slower.

Although... is there any mileage in making this `Async.each[Limit]` instead?

Reply:
I'm just a bit skittish about the extra function-call and callback overhead compared to the good old for-loop. We could certainly do an eachLimit to get a bit of concurrency.